### PR TITLE
test(accelerate): new dataproxy flavor

### PR DESCRIPTION
**Pull request by millsp**
Closes https://github.com/prisma/team-orm/issues/221

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -1094,6 +1094,7 @@ jobs:
     concurrency: dataproxy-${{ matrix.platform }}-${{ matrix.clientEngine }}
     env:
       FLAVOR: ${{ matrix.flavor }}
+      PRISMA_GENERATE_FLAG: ${{ matrix.flavor == 'DP1' && '--data-proxy' || '--accelerate' }}
```
Comment (janpio) [inline .github/workflows/test.yaml]: That would not really be necessary, as we know they work exactly the same. 

But: Nice.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -1087,26 +1087,34 @@ jobs:
           - nodejs-postgresql-logs
           - nodejs-mongodb-logs
           - nodejs-tracing
-        clientEngine: ['<dataproxy>'] # value for script logic
+        clientEngine: [<dataproxy>] # value for script logic
         os: [ubuntu-20.04] #, windows-latest, macos-latest]
     runs-on: ${{ matrix.os }}
-    concurrency: dataproxy-${{ matrix.platform }}-${{ matrix.clientEngine }}
+    concurrency:
+      group: dataproxy-${{ matrix.platform }}-${{ matrix.clientEngine }}
```
Comment (janpio) [inline .github/workflows/test.yaml]: I think this actually needs the flavor so jobs in the same workflow run do not cancel each other, but also across separate runs.

**Comment by millsp** [comment]
We should perhaps disable the extension tests for now, as it returns unexpected results sometimes.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -149,7 +149,7 @@ jobs:
 
   process-managers:
     needs: [detect_jobs_to_run]
-    if: ${{ fromJson(needs.detect_jobs_to_run.outputs.jobs)['process-managers'] == true }}
+    if: ${{ false && fromJson(needs.detect_jobs_to_run.outputs.jobs)['process-managers'] == true }}
```
Comment (janpio) [inline .github/workflows/test.yaml]: Why was this change needed?

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -1087,7 +1090,7 @@ jobs:
           - nodejs-postgresql-logs
           - nodejs-mongodb-logs
           - nodejs-tracing
-        clientEngine: ['<dataproxy>'] # value for script logic
+        clientEngine: [<dataproxy>] # value for script logic
```
Comment (janpio) [inline .github/workflows/test.yaml:1093]: What does this switch indicate?

File: dataproxy/cloudflare-workers-module/package.json [hunk 5]
```diff
@@ -5,7 +5,8 @@
     "test": "jest"
   },
   "dependencies": {
-    "@prisma/client": "4.17.0-dev.77"
+    "@prisma/client": "4.17.0-dev.77",
+    "@prisma/extension-accelerate": "0.3.1"
```
Comment (janpio) [inline dataproxy/cloudflare-workers-module/package.json:9]: How do we go about updating this? Renovate?

File: dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts [hunk 6]
```diff
@@ -1,7 +1,15 @@
...
 import util from 'util'
 
-const prisma = new PrismaClient()
+let prisma = new PrismaClient()
+let timeouts = { maxWait: 2_000, timeout: 5_000 }
+
+if (process.env.DATAPROXY_FLAVOR === 'DP2+Extension') {
+  prisma = prisma.$extends(withAccelerate()) as any
+  timeouts = { maxWait: 30_000, timeout: 50_000 }
+}
```
Comment (janpio) [inline dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts:11]: Ugh.

File: dataproxy/nodejs-cockroachdb-itx/test/lots-of-activities.test.ts [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
 import { PrismaClient } from '@prisma/client'
+import { withAccelerate } from '@prisma/extension-accelerate'
 import { config } from '../config'
 
 const activities = config['lots-of-activities'].amount
+let timeouts = { maxWait: 2_000, timeout: 5_000 }
+
+if (process.env.DATAPROXY_FLAVOR === 'DP2+Extension' || process.env.DATAPROXY_FLAVOR === 'DP2') {
+  timeouts = { maxWait: 30_000, timeout: 50_000 }
+}
```
Comment (janpio) [inline dataproxy/nodejs-cockroachdb-itx/test/lots-of-activities.test.ts:11]: Ugh 2.

File: dataproxy/nodejs-mongodb-itx/test/functional.test.ts [hunk 8]
```diff
@@ -1,7 +1,15 @@
...
 import util from 'util'
 
-const prisma = new PrismaClient()
+let prisma = new PrismaClient()
+let timeouts = { maxWait: 2_000, timeout: 5_000 }
+
+if (process.env.DATAPROXY_FLAVOR === 'DP2+Extension') {
+  prisma = prisma.$extends(withAccelerate()) as any
+  timeouts = { maxWait: 30_000, timeout: 50_000 }
+}
```
Comment (janpio) [inline dataproxy/nodejs-mongodb-itx/test/functional.test.ts:11]: Ugh 3

File: dataproxy/nodejs-mongodb-itx/test/lots-of-activities.test.ts [hunk 9]
```diff
@@ -1,14 +1,24 @@
...
 import { PrismaClient } from '@prisma/client'
+import { withAccelerate } from '@prisma/extension-accelerate'
 import { config } from '../config'
 
 const activities = config['lots-of-activities'].amount
+let timeouts = { maxWait: 2_000, timeout: 5_000 }
+
+if (process.env.DATAPROXY_FLAVOR === 'DP2+Extension' || process.env.DATAPROXY_FLAVOR === 'DP2') {
+  timeouts = { maxWait: 30_000, timeout: 50_000 }
+}
```
Comment (janpio) [inline dataproxy/nodejs-mongodb-itx/test/lots-of-activities.test.ts:11]: (Last Ugh comment I make... I assume it applies on all of these.)

**Review by janpio — COMMENTED** [review]
👍

File: .github/workflows/test.yaml [hunk 10]
```diff
@@ -1065,6 +1065,9 @@ jobs:
       - name: notify-slack
         if: failure()
         run: bash .github/slack/notify-failure.sh ${{ github.job }} ${{ matrix.platform }}
+
+  # TODO databases are shared between the steps (DP1, DP2, DP2+Accelerate), so we runt them in sequence
```
Comment (Jolg42) [inline .github/workflows/test.yaml]:
```suggestion
  # TODO databases are shared between the steps (DP1, DP2, DP2+Accelerate), so we run them in sequence
```

File: .github/workflows/test.yaml (hunk 3 above)
Comment (Jolg42) [reply .github/workflows/test.yaml]: I think this needs to be reverted and was only used to skip tests

File: .github/workflows/test.yaml [hunk 11]
```diff
@@ -1132,23 +1130,73 @@ jobs:
       - name: Install Dependencies
         run: pnpm install
 
-      - name: test ${{ matrix.platform }}
-        id: run-test
+      - name: test DP1 ${{ matrix.platform }}
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 60
```
Comment (Jolg42) [inline .github/workflows/test.yaml]:
```suggestion
          timeout_minutes: 15
```
Would be better, I think?

File: .github/workflows/test.yaml [hunk 12]
```diff
@@ -1132,23 +1130,73 @@ jobs:
...
+          ITX_PDP_MYSQL: ${{ secrets.ITX_PDP_MYSQL }}
+          ITX_PDP_POSTGRESQL_DATABASE_URL: ${{ secrets.ITX_PDP_POSTGRESQL_DATABASE_URL }}
+          ITX_PDP_POSTGRESQL: ${{ secrets.ITX_PDP_POSTGRESQL }}
+          ITX_PDP_COCKROACHDB_DATABASE_URL: ${{ secrets.ITX_PDP_COCKROACHDB_DATABASE_URL }}
+          ITX_PDP_COCKROACHDB: ${{ secrets.ITX_PDP_COCKROACHDB }}
+
+      - name: test DP2 ${{ matrix.platform }}
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 60
```
Comment (Jolg42) [inline .github/workflows/test.yaml]:
```suggestion
          timeout_minutes: 15
```
Would be better, I think?

File: .github/workflows/test.yaml [hunk 13]
```diff
@@ -1132,23 +1130,73 @@ jobs:
...
+          ITX_PDP_MYSQL: ${{ secrets.DP2_ITX_PDP_MYSQL }}
+          ITX_PDP_POSTGRESQL_DATABASE_URL: ${{ secrets.ITX_PDP_POSTGRESQL_DATABASE_URL }}
+          ITX_PDP_POSTGRESQL: ${{ secrets.DP2_ITX_PDP_POSTGRESQL }}
+          ITX_PDP_COCKROACHDB_DATABASE_URL: ${{ secrets.ITX_PDP_COCKROACHDB_DATABASE_URL }}
+          ITX_PDP_COCKROACHDB: ${{ secrets.DP2_ITX_PDP_COCKROACHDB }}
+
+      - name: test DP2 with Extension ${{ matrix.platform }}
         uses: nick-invision/retry@v2
         with:
           timeout_minutes: 60
```
Comment (Jolg42) [inline .github/workflows/test.yaml]:
```suggestion
          timeout_minutes: 15
```
Would be better, I think?

File: dataproxy/cloudflare-workers-module/package.json (hunk 5 above)
Comment (Jolg42) [reply dataproxy/cloudflare-workers-module/package.json:9]: Renovate would be good, I think for this package.

File: dataproxy/nodejs-postgresql-itx/test/lots-of-activities.test.ts [hunk 14]
```diff
@@ -1,14 +1,26 @@
...
 import { config } from '../config'
 
 const activities = config['lots-of-activities'].amount
+let timeouts = { maxWait: 2_000, timeout: 5_000 }
+
+if (process.env.DATAPROXY_FLAVOR === 'DP2+Extension' || process.env.DATAPROXY_FLAVOR === 'DP2') {
+  timeouts = { maxWait: 30_000, timeout: 50_000 }
+}
+
+console.log('timeouts', timeouts)
```
Comment (Jolg42) [inline dataproxy/nodejs-postgresql-itx/test/lots-of-activities.test.ts]:
```suggestion
```

File: dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts (hunk 6 above)
Comment (Jolg42) [reply dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts:11]: Is this is only needed in CI somehow?

**Comment by Jolg42** [comment]
@millsp it looks like some tests are failing, are they flaky?

**Comment by janpio** [comment]
What I see in the results:
- Deno+DP1 seems to have a new problem (Which means we do not even run the DP2 tests)
- DP2 might have some problems with iTX

**Review by Jolg42 — APPROVED** [review]
Nice one 💯 !
Unfortunate that some DP2 tests are not stable, but an interesting finding for sure!

See comments before merging 📜

File: .github/workflows/test.yaml (hunk 4 above)
Comment (millsp) [reply .github/workflows/test.yaml:1093]: Just a nit, they are equivalent

File: dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts (hunk 6 above)
Comment (millsp) [reply dataproxy/nodejs-cockroachdb-itx/test/functional.test.ts:11]: I could maybe drop it to around 15s which seems to be what it hovers at, but taking the risk to make it more flaky.
